File: java/src/main/java/com/hydrapayments/sdk/model/ApiResponse.java
package com.hydrapayments.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class ApiResponse<T> {

    private boolean success;
    private T data;
    private String error;

    public ApiResponse() {}

    public boolean isSuccess() { return success; }
    public void setSuccess(boolean success) { this.success = success; }

    public T getData() { return data; }
    public void setData(T data) { this.data = data; }

    public String getError() { return error; }
    public void setError(String error) { this.error = error; }
}

File: dotnet/src/HydraPayments.Sdk/Models/ApiResponse.cs
using System.Text.Json.Serialization;

namespace HydraPayments.Sdk.Models;

public class ApiResponse<T>
{
    [JsonPropertyName("success")]
    public bool Success { get; set; }

    [JsonPropertyName("data")]
    public T? Data { get; set; }

    [JsonPropertyName("error")]
    public string? Error { get; set; }
}
